scripts/watermarkstat

```diff
@@ -177,28 +174,47 @@ class Watermarkstat(object):
 
         return pg_index
 
+    def build_header(self, wm_type):
+        if wm_type is None:
+            print >> sys.stderr, "Header info is not available!"
+            sys.exit(1)
+
+        self.header_list = ['Port']
+        header_map = wm_type["obj_map"]
+        single_key = header_map.keys()[0]
+        header_len = len(header_map[single_key])
+        min_idx = float("inf")
+
+        for name, counter_oid in header_map[single_key].items():
+            curr_idx = int(wm_type["idx_func"](counter_oid))
+            min_idx = min(min_idx, curr_idx)
+
+        self.min_idx = int(min_idx)
+        self.header_list += ["{}{}".format(wm_type["header_prefix"], idx) for idx in range(self.min_idx, self.min_idx + header_len)]
+
     def get_counters(self, table_prefix, port_obj, idx_func, watermark):
         """
             Get the counters from specific table.
         """
 
-        fields = ["0"] * 8
+        fields = ["0"]* (len(self.header_list) - 1)
```

Comment:
Worth a comment here to explain why '- 1'?

Reply:
Done!